File: Server/BattleServer/Battle/FiniteStateMachine/EntityStateContext.cs
using Core.FiniteStateMachine;
using Google.Protobuf;

namespace BattleServer.Battle.FiniteStateMachine
{
	public class EntityStateContext
	{
		public int shakeTime;
		public int skillID;

		public void DecodeSnapshot( CodedInputStream reader )
		{
			this.shakeTime = reader.ReadInt32();
			this.skillID = reader.ReadInt32();
		}
	}
}

File: Client/src/Model/Logic/FSM/EntityStateContext.ts
import * as $protobuf from "../../../Libs/protobufjs";
import { ISnapshotable } from "../../ISnapshotable";

export class EntityStateContext implements ISnapshotable {
	public shakeTime: number;
	public skillID: number;

	public EncodeSnapshot(writer: $protobuf.Writer | $protobuf.BufferWriter): void {
		writer.int32(this.shakeTime);
		writer.int32(this.skillID);
	}

	public DecodeSnapshot(reader: $protobuf.Reader | $protobuf.BufferReader): void {
		this.shakeTime = reader.int32();
		this.skillID = reader.int32();
	}
}